### ISSUE TITLE
BUG: BackdropFilter is inconsistent across the edges when using with a webview

### ISSUE THREAD

**Issue by zhushenwudi** [flutter/flutter#120028]
issue bind #114740
@AlexDochioiu
I'm sure this bug still exists so far

[✓] Flutter (Channel master, 3.8.0-4.0.pre.23, on macOS 13.0 22A380 darwin-arm64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[!] Android Studio
    ✗ Unable to find bundled Java version.
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.3)
[✓] VS Code (version 1.75.0)
[✓] Connected device (3 available)
    ! Error: iPhone is busy: Fetching debug symbols for iPhone. Xcode will continue when iPhone is finished. (code -10)
[✓] HTTP Host Availability

I am only tested on iOS.
i use a we_slide library, it has a suspended layer with backdropfilter container. And I write a stack in the suspended  layer, main view contain a webview with flutter_inappwebview.  When the suspended layer opened, there is a webview rect area will without blur effect

**Comment by zhushenwudi** [comment]
webview container area:
![image](https://user-images.githubusercontent.com/30340027/216803682-e35c9c14-9e11-4120-9ee1-e44d2e86cc11.png)

When the floating suspended layer is displayed on the non-webview, the effect is as follows

![image](https://user-images.githubusercontent.com/30340027/216803804-1ce2e081-4bea-4c20-a2ed-f8624cc1c6d3.png)

When the floating suspended layer is above WebView, the effect is as follows

![image](https://user-images.githubusercontent.com/30340027/216803702-df566c8a-78a3-4ffb-a0d3-43da814ae76a.png)

**Comment by moffatman** [comment]
I think it would be fixed with my change here: https://github.com/flutter/engine/pull/39244. Do you have an example program I can run to try it?

**Comment by zhushenwudi** [comment]
> flutter/engine#39244

Ok, I'll make a demo for testing

**Comment by zhushenwudi** [comment]
> I think it would be fixed with my change here: [flutter/engine#39244](https://github.com/flutter/engine/pull/39244). Do you have an example program I can run to try it?

hello again, i make a demo for testing:

https://github.com/zhushenwudi/BackdropFilterDemo

**Comment by zhushenwudi** [comment]
This bug effect is very interesting

https://user-images.githubusercontent.com/30340027/216876065-8bf3c9ce-72f8-4702-89ad-1cdc7db4253a.mp4

**Comment by moffatman** [comment]
Ok, so your demo isn't affected by my bug fix. This is some separate problem. 

It looks like there is some limitation in the platform view blur implementation where we can't blur easily across the edges of the webview. You can see it easily by reducing sigmaY, the affected area shrinks. 

I found this related [article](https://medium.com/@imho_ios/avoid-artifacts-on-uiblureffect-edges-c30e737c21fb), but I couldn't find a way to implement the advice (not an expert of UIKit). 

It seems like a hard issue to get right. Tag @cyanglaz for opinion.

**Comment by exaby73** [comment]
I can reproduce this issue with the sample you provided. @zhushenwudi is it possible for you to modify the code sample to exclude any 3rd party libraries, eg: `we_slide` and using `webview_flutter` instead of `webview_flutter_plus`?

**Comment by zhushenwudi** [comment]
> webview_flutter

I have tested webview_flutter, it does not support WebView with params
and, I also tested flutter_inappwebview, it can open webview with params, but it has the same effect with flutter_webview_plus, due to this library is very heavy, last I only choose webview_flutter_plus

**Comment by zhushenwudi** [comment]
if you know how to open an local asset resouce webview with params, please write it, i will test, thx

**Comment by exaby73** [comment]
Is it reproducible with loading a URL, like https://flutter.dev for example?

**Comment by zhushenwudi** [comment]
No, it can't be loaded because it doesn't meet the ios specification by directly splicing the URL

**Comment by zhushenwudi** [comment]
I tested the official webview_flutter is still the current effect, without any progress

```dart
WebViewController controller = WebViewController();

  @override
  void initState() {
    controller
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..setBackgroundColor(Colors.greenAccent)
      ..loadFlutterAsset("assets/www/index.html");
    super.initState();
  }
...
child: WebViewWidget(controller: controller)
```

**Comment by zhushenwudi** [comment]
and when i append params, it will be throw a exception, such as:

[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(FWFURLParsingError, Failed parsing file path., Initializing NSURL with the supplied 'Frameworks/App.framework/flutter_assets/assets/www/index.html?id=1' path resulted in a nil value., null)
#0      WKWebViewHostApi.loadFlutterAsset (package:webview_flutter_wkwebview/src/common/web_kit.g.dart:2202:7)

i dont know how to append params

**Comment by cyanglaz** [comment]
The [Possible Future Update: Dividing Line](https://docs.google.com/document/d/1V7Jc_RGaknrBBPPBBKB8lT7f3PKhYr8sin35MSMFAf4/edit#heading=h.qbklsl1nwn2g) explains why the edge of the PlatformViews are not blurred and the some approaches to solve this. This is not something currently worked on by the team. It would be interesting if someone could try it out and see the results of these approaches.

**Comment by exaby73** [comment]
Since I can reproduce this issue with the steps highlighted in https://github.com/flutter/flutter/issues/120028#issuecomment-1418438680, I'll label this issue

**Comment by Reprevise** [comment]
another way to reproduce the issue: https://github.com/Reprevise/flutter_ios_platform_view_blur_issue

**Comment by sanalkv** [comment]
Facing the same issue here. Not able to use BackdropFilter on top of webview.

**Comment by marianhlavac** [comment]
Any updates to this?

**Comment by wz-wenchao** [comment]
same problem, only happened on ios

I was use Flutter stable channel 3.29.2
webview_flutter: ^4.10.0

the code is simple

```
import 'dart:ui';

import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(home: HomePage());
  }
}

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<HomePage> {
  late WebViewController controller;

  @override
  void initState() {
    super.initState();

    controller =
        WebViewController()
          ..setBackgroundColor(Colors.transparent)
          ..loadRequest(Uri.parse("https://flutter.dev"));
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text("Demo"),
      ),
      body: Stack(
        children: [
          WebViewWidget(controller: controller),
          Center(
            child: ClipRRect(
              borderRadius: BorderRadius.circular(50),
              child: BackdropFilter(
                filter: ImageFilter.blur(sigmaX: 10, sigmaY: 10),
                child: Container(
                  color: Colors.black.withValues(alpha: 0.5),
                  child: SizedBox(width: 180, height: 180),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
```

![Image](https://github.com/user-attachments/assets/27cdbaa3-742b-48e6-ab44-fc5818400abc)

as you see, the corner is not clip well
